### ISSUE TITLE
osprey.printGpuInfo() Cuda error

### ISSUE THREAD

**Issue by hnisonoff** [donaldlab/OSPREY3#47]
I just installed the python version of OSPREY on my laptop (2015 era Macbook Pro). When I went to print the GPU info I received the following error:

```
In [2]: import osprey

In [3]: osprey.start()
OSPREY 3.0-alpha2

In [4]: osprey.printGpuInfo()
Discovering CUDA GPUs... none found
---------------------------------------------------------------------------
jcuda.CudaExceptionPyRaisable             Traceback (most recent call last)
<ipython-input-4-c610ee56c954> in <module>()
----> 1 osprey.printGpuInfo()

/Users/hunternisonoff/anaconda3/lib/python3.6/site-packages/osprey/__init__.py in printGpuInfo()
    167         Both Cuda and OpenCL APIs are queried.
    168 	'''
--> 169         c.gpu.cuda.Diagnostics.main(None)
    170         c.gpu.opencl.Diagnostics.main(None)
    171

jcuda.CudaExceptionPyRaisable: jcuda.CudaException: CUDA_ERROR_NOT_INITIALIZED
```

I am pretty ignorant of Cuda and GPUs at the moment. Any hints for how to debug this?

**Comment by markhallen369** [comment]
If you go to the Apple menu —> About This Mac it should say what graphics hardware you have.  I think MacBook Pro’s don’t have Cuda-capable GPUs built in, at least mine doesn’t (it has an Intel GPU but Cuda only runs on Nvidia GPU’s).  

Mark

**Comment by hnisonoff** [comment]
Thanks Mark. I would expect this to not crash the program however, as it is just checking for GPUs. I am not asking OSPREY to run Cuda.

**Comment by markhallen369** [comment]
Yeah I agree, can we maybe catch this exception?
I also have a problem with the latest version on a Mac—the pythonDevelop command crashes and says that it can’t find an appropriate JPype, even though I already installed the provided JPype wheel with pip:

mhall44s-MacBook-Pro:OSPREY_refactor mhall44$ ./gradlew pythonDevelop

> Task :pythonDevelop
Ignoring indexes: https://pypi.python.org/simple/
Obtaining file:///Users/mhall44/Mark/OSPREY_refactor/python
  Running setup.py (path:/Users/mhall44/Mark/OSPREY_refactor/python/setup.py) egg_info for package from file:///Users/mhall44/Mark/OSPREY_refactor/python
    /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    
Downloading/unpacking JPype-py2>=0.5.8 (from osprey==3.0-alpha4)
  Could not find any downloads that satisfy the requirement JPype-py2>=0.5.8 (from osprey==3.0-alpha4)
Cleaning up...
No distributions at all found for JPype-py2>=0.5.8 (from osprey==3.0-alpha4)
Storing debug log for failure in /Users/mhall44/.pip/pip.log


FAILURE: Build failed with an exception.

Mark

**Comment by cuchaz** [comment]
Hi guys,

Yeah, we should catch that CUDA exception. Last time I checked, that exception wasn't thrown on my laptop (which I think doesn't have CUDA installed), so it might be difficult for me to debug this. I just tried running printGpuInfo() on my laptop now and I actually got a different error, so it's clear we need to spend more time on this anyway.

As for the jpype problems, we should split that into a separate issue on Github.
